### PR TITLE
4366 - Date validation is inconsistent between inline and admin check(backend)

### LINKED ISSUE #4366: [Bug] Date validation is inconsistent between inline and admin check(backend)
### Description
Date validation is happening between the in-line and admin check is inconsistent. When the admin check is enabled the validation in the page appears to work as expected. The admin check panel:
* Doesn't catch when Start Dates are later than End Dates
* Doesn't validate when an invalid year is entered

### Steps to reproduce

1. Login and view an APD. Turn on Admin Check
2. Navigate to Activity Schedule
3. Enter dates and observe the above issues

**Expected results-**
* Dates validate consistently between admin check panel and inline validation

**Actual results-**

### Role Affected

- [x] All
- [ ] Federal Admin
- [ ] State Admin
- [ ] State Staff
- [ ] State Contractor
- [ ] System Admin

### Environment

- [x] All
- [ ] PR #\_\_\_\_
- [ ] Staging
- [ ] Production

### Browser

- [x] All
- [ ] Firefox
- [ ] Chrome
- [ ] IE

**Cypress test created OR Ticket for cypress test created-**

**Related user story/bug-** (if applicable)

**Additional information-** (optional)

**Logs or error output/Screenshot/Video-** (at least 1 of these)

### PR DESCRIPTION
**Pull request by Sun-Mountain**
Resolves #4366 

### Description
Date validation is happening between the in-line and admin check is inconsistent. When the admin check is enabled the validation in the page appears to work as expected. The admin check panel:

- Doesn't catch when Start Dates are later than End Dates
- Doesn't validate when an invalid year is entered

### Chromatic Link
https://www.chromatic.com/build?appId=61d5b948cf6f17003a12bf77&number=1335

### Significant changes or possible side effects
Per discussion, valid years are 1960 thru 2151.

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| Date field with start and end date | admin check is on | should catch if end date is before start date with appropriate validation error | cypress |
| Date field | admin check is one | should catch an invalid year with appropriate validation error | cypress |

### Steps to manually verify this change

1. Login and view an APD. Turn on Admin Check
2. Navigate to Activity Schedule
3. Enter dates

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [ ] Chromatic link added above
- [x] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [x] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when

- [x] Code has been reviewed by someone other than the original author

### This pull request is ready to review when QA has

- [x] Verified the functionality related to the change
- [x] Verified that the change works with Narrator on Windows
- [x] Verified that the change works with VoiceOver on Mac
- [x] Verified all updated pages with the WAVE tool
- [x] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [x] Product has approved the experience

### CODE REVIEW

File: web/src/components/DateField.js [hunk 1]
```diff
@@ -38,6 +34,8 @@ const DateField = ({
     yearInvalid: false
   });
 
+  console.log({ value });
```
Comment (thetif) [inline web/src/components/DateField.js:37]: remove console log

File: web/src/components/DateField.js [hunk 2]
```diff
@@ -46,9 +44,9 @@ const DateField = ({
       const lastDayOfMonth = new Date(parseInt(year), parseInt(month) - 1, 0);
       setInvalidObject({
         dayInvalid:
-          !isNumeric(day) || +day < 1 || +day > lastDayOfMonth.getDate() + 1,
-        monthInvalid: !isNumeric(month) || +month < 1 || +month > 12,
-        yearInvalid: !isNumeric(year) || +year < 1900 || +year > 2100
+          !isNumeric(day) || day < 1 || day > lastDayOfMonth.getDate() + 1,
```
Comment (thetif) [inline web/src/components/DateField.js:47]: the +day or whatever will convert the string to a number, so you probably still need it here

File: web/src/components/DateField.js [hunk 3]
```diff
@@ -16,13 +17,8 @@ const dateParts = value => {
...
 
 const formatDate = ({ day = '', month = '', year = '' } = {}) => {
-  if (day === '' && month === '' && year === '') {
-    return '';
-  }
   // Make sure it's an ISO-8601 date, which uses 2-digit month and day
-  return `${year}-${month < 10 ? `0${month}` : month}-${
-    day < 10 ? `0${day}` : day
-  }`;
+  return format(new Date(year, month - 1, day), 'yyyy-MM-dd');
```
Comment (tbolt) [inline web/src/components/DateField.js:21]: nice, this is way cleaner

**Review by tbolt — APPROVED** [review]
Looks great

**Review by jeromeleecms — APPROVED** [review]
This works as described. We will be re-enabling inline validating in 4661 for the state date.

**Pull request merged**